py/torch_tensorrt/dynamo/conversion/impl/shuffle.py

```diff
@@ -61,37 +60,113 @@ def pixel_shuffle(
     input: TRTTensor,
     upscale_factor: int,
 ) -> TRTTensor:
-    shape = input.shape
-    in_channels, in_height, in_width = shape[-3:]
-    out_channels = in_channels // (upscale_factor**2)
-    out_height = in_height * upscale_factor
-    out_width = in_width * upscale_factor
-    new_shape = shape[:-3] + (
-        out_channels,
+    # Get input shape tensor
+    input_shape_tensor = ctx.net.add_shape(input).get_output(0)
+
+    # Extract in_channels, in_height, and in_width from the input shape tensor
+    in_channels_tensor = ctx.net.add_slice(
+        input_shape_tensor, start=(len(input.shape) - 3,), shape=(1,), stride=(1,)
+    ).get_output(0)
+    in_height_tensor = ctx.net.add_slice(
+        input_shape_tensor, start=(len(input.shape) - 2,), shape=(1,), stride=(1,)
+    ).get_output(0)
+    in_width_tensor = ctx.net.add_slice(
+        input_shape_tensor, start=(len(input.shape) - 1,), shape=(1,), stride=(1,)
+    ).get_output(0)
+
+    # Calculate out_channels, out_height, and out_width as tensors
+    upscale_factor_sq = upscale_factor * upscale_factor
+    upscale_factor_sq_tensor = get_trt_tensor(
+        ctx, upscale_factor_sq, f"{name}_upscale_factor_sq"
+    )
+
+    out_channels_tensor = impl.elementwise.div(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_channels_tensor",
+        in_channels_tensor,
+        upscale_factor_sq_tensor,
+    )
+    out_height_tensor = impl.elementwise.mul(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_height_tensor",
+        in_height_tensor,
         upscale_factor,
+    )
+    out_width_tensor = impl.elementwise.mul(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_width_tensor",
+        in_width_tensor,
         upscale_factor,
-        in_height,
-        in_width,
     )
-    reshaped_tensor = reshape(
-        ctx, target, source_ir, f"{name}_reshape1", input, new_shape
+
+    # Construct new shape tensor
+    new_shape_tensors = [
+        ctx.net.add_slice(
+            input_shape_tensor, start=(i,), shape=(1,), stride=(1,)
+        ).get_output(0)
+        for i in range(len(input.shape) - 3)
+    ]
+    new_shape_tensors += [
+        out_channels_tensor,
+        get_trt_tensor(ctx, upscale_factor, f"{name}_upscale_factor_1"),
+        get_trt_tensor(ctx, upscale_factor, f"{name}_upscale_factor_2"),
+        in_height_tensor,
+        in_width_tensor,
+    ]
+
+    new_shape_tensor = impl.cat.cat(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_new_shape_tensor",
+        new_shape_tensors,
+        0,
+        np.int32,
     )
-    rank = len(shape)
+
+    # Reshape tensor
+    reshape_layer = ctx.net.add_shuffle(input)
+    reshape_layer.set_input(1, new_shape_tensor)
+    reshaped_tensor = reshape_layer.get_output(0)
```

Comment:
Is it possible to reuse `impl.shuffle.reshape`?

Reply:
Thank you for your comment. I have updated the code to reuse the `reshape` function. This makes the code simpler and more consistent.